File: LeetCode-Programs/Two Sum II - Input Array Is Sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        new_list=[]
        i, j =0, len(numbers)-1
        while i<j:
            if numbers[i]+numbers[j]<target:
                i=i+1
            elif numbers[i]+numbers[j]>target:
                j=j-1
            else:
                new_list.append(i+1)
                new_list.append(j+1)
                return new_list
        return new_list
      
      
      
      
      
#TLE

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        new_list=[]
        i, j =0, len(numbers)-1
        while i<j:
            if numbers[i]+numbers[j]<target:
                i=i+1
            elif numbers[i]+numbers[j]>target:
                j=j-1
            else:
                new_list.append(i+1)
                new_list.append(j+1)
                
        return new_list
      
      
      
      
      

File: LeetCode-Programs/Two Sum II - Input Array Is Sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> vec;
        int i=0, j=numbers.size()-1;
        while(i<j)
        {
            if(numbers[i]+numbers[j]<target)
                i++;
            else if(numbers[i]+numbers[j]>target)
                j--;
            else
            {
                vec.push_back(i+1);
                vec.push_back(j+1);
                return vec;
            }
        }
        return vec;
    }
};
